### PR TITLE
TimeDimensionDisplay DST day conversion fix

### PR DESCRIPTION
**Pull request by egor-ryashin**
## Checklist
- [x] Manual verification
- [ ] Unit test coverage
- [ ] E2E test coverage
- [x] Needs manual QA?

#### Issue addressed: 

https://github.com/rilldata/rill/issues/3560#issuecomment-1881384653
## Steps to Verify

1. Open 311 Operational Metrics dashboard.
1. Set `first_day_of_week: 7`
1. In Chrome Developer Tools -> Sensors modify Location to `America/New_York` Timezone ID.
1. Drill into Median Time to Resolve Ticket (Hours) from main dashboard view
1. Adjust date range to Oct 29 - Nov 30 and make sure that metric trends is set to week
1. Check that the column header in the TDD shows Nov 5 (Sun) as first day of week.
1. Select time subrange ending with Nov 5 by dragging the mouse over the chart.
1. Check the subrange control shows Nov 5.

image

### CODE REVIEW

File: web-common/src/features/dashboards/time-series/utils.ts [hunk 1]
```diff
@@ -44,20 +43,19 @@ export function prepareTimeSeries(
   original,
   comparison,
   timeGrainDuration: string,
-  zone: string,
 ) {
   return original?.map((originalPt, i) => {
     const comparisonPt = comparison?.[i];
 
-    const ts = adjustOffsetForZone(originalPt.ts, zone);
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-series/utils.ts:52]: For the charts, we need to have this offset. Javascript assumes the dates are in the user's timezone. We need to account for that across our charts

**Review by djbarnwal — CHANGES_REQUESTED** [review]
This PR breaks charting at multiple places

image

The charts only works if the selected timezone is same as the user's timezone.

File: web-common/src/features/dashboards/time-series/utils.ts (hunk 1 above)
Comment (egor-ryashin) [reply web-common/src/features/dashboards/time-series/utils.ts:52]: could you elaborate what is broken in the screenshot?

**Comment by egor-ryashin** [comment]
The issue is more complex than I expected @djbarnwal is going to take a look

**Comment by egor-ryashin** [comment]
@djbarnwal I wonder if there're any updates?

**Comment by begelundmuller** [comment]
Closing as stale. We can reopen if there are further developments.

**Pull request closed without merging**